Source/FieldSolver/FiniteDifferenceSolver/MacroscopicProperties/MacroscopicProperties.H

```diff
@@ -2,34 +2,144 @@
 #define WARPX_MACROSCOPICPROPERTIES_H_
 
 
+#include "Parser/WarpXParserWrapper.H"
 #include <AMReX_MultiFab.H>
 
 
+
 /**
- * \brief This class contains the macroscopic parameters of the medium needed to
+ * \brief This class contains the macroscopic properties of the medium needed to
  * evaluate macroscopic Maxwell equation.
  */
 class
 MacroscopicProperties
 {
 public:
      MacroscopicProperties (); // constructor
-     /** \brief Read user-defined macroscopic properties. Called in constructor. */
+     /** Read user-defined macroscopic properties. Called in constructor. */
      void ReadParameters ();
-     /** return Real, sigma (conductivity) of the medium. */
-     amrex::Real sigma () const noexcept {return m_sigma;}
-     /** return Real, epsilon (permittivity) of the medium. */
-     amrex::Real epsilon () const noexcept {return m_epsilon;}
-     /** return Real, mu (permeability) of the medium. */
-     amrex::Real mu () const noexcept {return m_mu;}
+     /** Initialize multifabs storing macroscopic multifabs */
+     void InitData ();
+
+     /** return MultiFab, sigma (conductivity) of the medium. */
+     amrex::MultiFab& getsigma_mf  () {return (*m_sigma_mf);}
+     /** return MultiFab, epsilon (permittivity) of the medium. */
+     amrex::MultiFab& getepsilon_mf  () {return (*m_eps_mf);}
+     /** return MultiFab, mu (permeability) of the medium. */
+     amrex::MultiFab& getmu_mf  () {return (*m_mu_mf);}
+
+     /** Initializes the Multifabs storing macroscopic properties
+      *  with user-defined functions(x,y,z).
+      */
+     void InitializeMacroMultiFabUsingParser (amrex::MultiFab *macro_mf,
+                                  ParserWrapper<3> *macro_parser, int lev);
+     /** Gpu Vector with index type of the conductivity multifab */
+     amrex::Gpu::ManagedVector<int> sigma_IndexType;
+     /** Gpu Vector with index type of the permittivity multifab */
+     amrex::Gpu::ManagedVector<int> epsilon_IndexType;
+     /** Gpu Vector with index type of the permeability multifab */
+     amrex::Gpu::ManagedVector<int> mu_IndexType;
+     /** Gpu Vector with index type of the Ex multifab */
+     amrex::Gpu::ManagedVector<int> Ex_IndexType;
+     /** Gpu Vector with index type of the Ey multifab */
+     amrex::Gpu::ManagedVector<int> Ey_IndexType;
+     /** Gpu Vector with index type of the Ez multifab */
+     amrex::Gpu::ManagedVector<int> Ez_IndexType;
+     /** Gpu Vector with index type of coarsening ratio with default value (1,1,1) */
+     amrex::Gpu::ManagedVector<int> macro_cr_ratio;
 
 private:
      /** Conductivity, sigma, of the medium */
-     amrex::Real m_sigma;
+     amrex::Real m_sigma = 0.0;
      /** Permittivity, epsilon, of the medium */
-     amrex::Real m_epsilon;
+     amrex::Real m_epsilon = 8.8541878128e-12;
```

Comment:
This should be the ep0 from `Utils/WarpXConst.H` for consistency. And the same below for `mu0`.

Reply:
Added this. Thank you!